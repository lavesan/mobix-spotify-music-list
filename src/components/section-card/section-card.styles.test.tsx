import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import 'jest-styled-components';
import { StyledSectionCard } from '.';

describe('Tests `StyledSectionCard` ', () => {
  let tree: ReactTestRendererJSON | null;

  beforeAll(() => {
    tree = renderer.create(<StyledSectionCard />).toJSON();
  })

  test('Matches snapshot ', () => {
    expect(tree).toMatchSnapshot();
  })
  
  test('Have styles ', () => {
    expect(tree).toHaveStyleRule('box-shadow', '0 0 6px #c2c2c2')
    expect(tree).toHaveStyleRule('width', '80%')
    expect(tree).toHaveStyleRule('background-color', '#fff')
    expect(tree).toHaveStyleRule('padding', '10px 50px')
    expect(tree).toHaveStyleRule('padding', '10px 20px', {
      media: '(max-width: 570px)',
    })
  })
})
