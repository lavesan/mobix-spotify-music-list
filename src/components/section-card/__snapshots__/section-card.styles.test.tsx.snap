// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests \`StyledSectionCard\`  Matches snapshot  1`] = `
.c0 {
  box-shadow: 0 0 6px #c2c2c2;
  border-radius: 5px;
  padding: 10px 50px;
  background-color: #fff;
  width: 80%;
}

@media (max-width:570px) {
  .c0 {
    padding: 10px 20px;
  }
}

<section
  className="c0"
/>
`;
